lib/archethic/oracle_chain/scheduler.ex

```diff
@@ -140,20 +140,59 @@ defmodule Archethic.OracleChain.Scheduler do
 
     index = Map.fetch!(indexes, summary_date)
 
-    if trigger_node?(summary_date, index + 1) do
-      new_oracle_data =
-        summary_date
-        |> Crypto.derive_oracle_address(index)
-        |> get_oracle_data()
-        |> Services.fetch_new_data()
+    new_oracle_data =
+      summary_date
+      |> Crypto.derive_oracle_address(index)
+      |> get_oracle_data()
+      |> Services.fetch_new_data()
+
+    {prev_pub, prev_pv} = Crypto.derive_oracle_keypair(summary_date, index)
+
+    {next_pub, _} = Crypto.derive_oracle_keypair(summary_date, index + 1)
 
+    tx =
+      Transaction.new_with_keys(
+        :oracle,
+        %TransactionData{
+          content: Jason.encode!(new_oracle_data),
+          code: ~S"""
+          condition inherit: [
+            # We need to ensure the type stays consistent
+            # So we can apply specific rules during the transaction validation
+            type: in?([oracle, oracle_summary]),
+
+            # We discard the content and code verification
+            content: true,
+
+            # We ensure the code stay the same
+            code: if type == oracle_summary do
+              regex_match?("condition inherit: \\[[\\s].*content: \\\"\\\"[\\s].*]")
+            else
+              previous.code
+            end
+          ]
+          """
+        },
+        prev_pv,
+        prev_pub,
+        next_pub
+      )
+
+    if trigger_node?(summary_date, index + 1) do
       if Enum.empty?(new_oracle_data) do
         Logger.debug("Oracle transaction skipped - no new data")
       else
-        send_polling_transaction(new_oracle_data, index, summary_date)
+        send_polling_transaction(tx)
       end
     else
-      Logger.debug("Oracle transaction skipped - not the trigger node")
+      DetectNodeResponsiveness.start_link(tx.address, fn count ->
+        if Enum.empty?(new_oracle_data) and trigger_node?(summary_date, index + 1, count) do
```

Comment:
You can reuse the emptiness check before the trigger node detection to avoid multiple calls

Reply:
OK, but generally 
` if a==b and c==d ` # second condition check happens only after first is true unlike in case of `if a==b or c==d` where both the conditions are checked. Is this not the case in elixir ?

Reply:
It does.
But you can simply:
- check if need to generate a new transaction (emptiness check)
- if yes
   - build tx 
   - if trigger node
        - send tx
    - if not trigger node
        - start watcher
- if no
   - skip